src/System.IO.Compression.Brotli/README.md

```diff
@@ -0,0 +1,214 @@
+        public static TransformationStatus Decompress(ReadOnlySpan<byte> source, Span<byte> destination, out int bytesConsumed, out int bytesWritten, ref Brotli.State state);
+        public static TransformationStatus FlushEncoder(ReadOnlySpan<byte> source, Span<byte> destination, out int bytesConsumed, out int bytesWritten, ref Brotli.State state, bool isFinished=true);
+        public static int GetMaximumCompressedSize(int inputSize);
+        public struct State : IDisposable {
+            public bool CompressMode { get; }
+            public void Dispose();
+            public void SetQuality(uint quality);
+            public void SetWindow(uint window);
+        }
+    }
+```
+```out bytesConsumed``` - number of bytes used from source data while executing the operation
+
+```out bytesWritten``` - number of bytes are written in destination while executing the operation
+
+```State``` is used in both compress and decompress mode to save a temporary status and data of process.  State will set automatically on initial call of Compress or Decompress. The state cannot be re-used for multiple data streams and cannot switch modes (compress vs. decompress).
+
+The ```Compress``` performs data compression. Taking data from ```source``` and writing compressed data to ```destination```.
+
+The ```FlushEncoder``` returns compressed data, which was sent to ```state``` using Compress. This method should always be called after all Compress executions (```isFinished = true```) or when you want to get compressed data immediately (```isFinished = false```).
+
+The ```Decompress``` Decompresses the data from ```source``` into ```destination```. 
+
+```SetQuality``` allows you to set quality of compression ```0 to 11```. The higher quality means the higher compression ratio, but more compute time.
+
+```SetWindow``` - Logarithm of recommended sliding LZ77 window size. Encoder may reduce this value, e.g. if input is much smaller than window size. Window size is (1 << value) - 16. Possible values: ```11 to 24```
+
+### Examples
+Simple method to compress bytes to file.
+```C#
+
+ static void CompressSimple(byte[] bytes, string OutFile)
+ {
+    Brotli.State state = new Brotli.State();
+    byte[] compressed = new byte[Brotli.GetMaximumCompressedSize(bytes.Length)];
+    state.SetQuality();
+    TransformationStatus result = Brotli.Compress(bytes, compressed, out int consumed, out int written, ref state);
+    while (result != TransformationStatus.Done)
+    {
+        result = Brotli.Compress(bytes, compressed, out consumed, out written, ref state);
+        bytes = bytes.AsSpan().Slice(consumed).ToArray();
+    }
+    result = Brotli.FlushEncoder(Array.Empty<byte>(), compressed, out consumed, out written, ref state);
+    byte[] ans = new byte[written];
+    Array.Copy(compressed, ans, written);
```

Comment:
I understand. However, imo, we should make sure that sample code showcases the patterns we expect users to write (which includes using spans where possible to avoid unnecessary work). Up to you :)